### PR TITLE
Find subnet ids from AZ

### PR DESCRIPTION
**Pull request by SimonRichardson**
### Checklist

 - [x] Checked if it requires a [pylibjuju](https://github.com/juju/python-libjuju) change?
 - [x] Added [integration tests](https://github.com/juju/juju/tree/develop/tests) for the PR?
 - [x] Added or updated [doc.go](https://discourse.jujucharms.com/t/readme-in-packages/451) related to packages changed?
 - [x] Do comments answer the question of why design decisions were made?

----

## Description of change

The following is a mechanical change to move the ability to find subnet
ids from available zones. This is a very generic function, one that can
be extracted to core and be tested in isolation.

## QA steps

Bootstrapping to aws should be fine.

```sh
juju bootstrap aws test --no-gui --debug
```

### CODE REVIEW

**Comment by SimonRichardson** [comment]
https://trello.com/c/QjbeiJtu/2302-mechanical-change-to-find-subnets-by-az

File: core/network/subnet_test.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	gc "gopkg.in/check.v1"
+)
+
+type subnetSuite struct {
+	testing.IsolationSuite
+}
+
+var _ = gc.Suite(&subnetSuite{})
+
+func (*PortRangeSuite) TestFindSubnetIDsForAZ(c *gc.C) {
```
Comment (manadart) [inline core/network/subnet_test.go]: Wrong suite.

File: core/network/subnet_test.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
+// Copyright 2019 Canonical Ltd.
```
Comment (manadart) [inline core/network/subnet_test.go]: 2020?

**Comment by SimonRichardson** [comment]
$$merge$$

**Pull request merged**